### PR TITLE
Bug fix and cleaning

### PR DESCRIPTION
**Pull request by ycantin**
Bug fix :
Silent out of bound access in checkIfJointsWithinBounds wreak havoc later (random segfault/memory corruption) : 
joint->getFirstVariableIndex() don't match and exceed new_jt_traj.points[0].velocities indexes when there is more joints in the robot model than in the move_group used.

There's some access check (try/catch std::out_of_range) but they don't work with [], they need .at() accessors.
I'm wondering if hardening the code ([] to .at()) is worth the effort.

Cleanup :
Some bits gathered when debugging, see if that's ok for you.

### CODE REVIEW

**Comment by AndyZe** [comment]
Can you roll back your work-in-progress commit and do a separate PR? That will make it easier on me.

This generally looks good but I'll have to do a more thorough review tomorrow.

**Comment by ycantin** [comment]
Do you mean one PR per commit, or an all-in-on commit ? (I'm not very familiar with git workflow)

**Comment by AndyZe** [comment]
A new PR for your passive joint filtering. It looks unrelated to the other
bug fixes.

File: jog_arm/src/jog_arm/jog_arm_server.cpp [hunk 1]
```diff
@@ -573,10 +573,10 @@ void JogCalcs::lowPassFilterPositions() {
 void JogCalcs::lowPassFilterVelocities(const Eigen::VectorXd &joint_vel) {
   for (size_t i = 0; i < jt_state_.name.size(); ++i) {
     jt_state_.velocity[i] =
-        velocity_filters_[i].filter(joint_vel[static_cast<long>(i)]);
+        velocity_filters_[i].filter(joint_vel[i]);
```
Comment (AndyZe) [inline jog_arm/src/jog_arm/jog_arm_server.cpp:576]: Indeed, these static casts aren't necessary, but machinekoder added them for a reason. See https://github.com/UTNuclearRoboticsPublic/jog_arm/pull/10

File: jog_arm/include/jog_arm/jog_arm_server.h [hunk 2]
```diff
@@ -249,7 +249,7 @@ class JogCalcs
                                        const Eigen::MatrixXd& new_jacobian,
                                        trajectory_msgs::JointTrajectory& new_jt_traj);
 
-  bool checkIfImminentCollision(jog_arm_shared& shared_variables, trajectory_msgs::JointTrajectory& new_jt_traj);
+  bool checkIfImminentCollision(jog_arm_shared& shared_variables);
```
Comment (AndyZe) [inline jog_arm/include/jog_arm/jog_arm_server.h:254]: Good catch!

**Comment by AndyZe** [comment]
This looks great up to **Remove unused checkIfImminentCollision parameter 59ecdbf**. I appreciate your attention to detail.  However, I definitely don't want to merge 23bd830. I'm on the fence about 
ea94279 and wanted to ask some more questions.

So, I'm planning to close this PR. Can you please make a new branch with the commits I want to keep:
git checkout 59ecdbf
git checkout -b ycantin-bug-fixes

Then make a new PR from that branch. I would definitely accept it.

**Comment by AndyZe** [comment]
@ycantin Re. filtering the passive joints out of the trajectory message: did you see any performance issues on your robot? Or what was the motivation?

It seems like there wouldn't be any harm with leaving the passive joints in the message, but I don't really have a way to test it.

Thanks a lot for all the bug fixes.

**Comment by ycantin** [comment]
I work with 6 3DOF legs and ikfast, but i was not able to get MoveIt to work with 3dof only (any hint ?), so ended adding 3 passive joints to the model to enable transform6D IK.

The other way around would be to consider these added joints as real, and discard them on the controller's level, but this mean doubling the size of all ros messages.

Moveit filters the passive joints before sending the final trajectory, so i wanted to do the same in jog_arm.

**Pull request closed without merging**